src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/dllmain.cpp

```diff
@@ -352,13 +352,91 @@ void RemoveHooks()
 HRESULT ExtRoLoadCatalog()
 {
     WCHAR filePath[MAX_PATH];
-    GetModuleFileNameW(nullptr, filePath, _countof(filePath));
+    if (!GetModuleFileNameW(nullptr, filePath, _countof(filePath)))
+    {
+        return HRESULT_FROM_WIN32(GetLastError());
+    }
     std::wstring manifestPath(filePath);
-    manifestPath += L".manifest";
+    HANDLE hActCtx = INVALID_HANDLE_VALUE;
+    auto exit = wil::scope_exit([&]
+    {
+        if (hActCtx != INVALID_HANDLE_VALUE)
+        {
+            ReleaseActCtx(hActCtx);
+        }
+    });
+    wil::unique_hmodule exeModule;
+    RETURN_IF_WIN32_BOOL_FALSE(GetModuleHandleExW(0, nullptr, &exeModule));
+    ACTCTXW acw = { sizeof(acw) };
+    acw.lpSource = manifestPath.c_str();
+    acw.hModule = exeModule.get();
+    acw.lpResourceName = MAKEINTRESOURCEW(1);
+    acw.dwFlags = ACTCTX_FLAG_HMODULE_VALID | ACTCTX_FLAG_RESOURCE_NAME_VALID;
+
+    hActCtx = CreateActCtxW(&acw);
+    if (hActCtx == INVALID_HANDLE_VALUE)
+    {
+        SetLastError(ERROR_OUTOFMEMORY);
+        return HRESULT_FROM_WIN32(GetLastError());
+    }
 
-    return WinRTLoadComponent(manifestPath.c_str());
-}
+    PACTIVATION_CONTEXT_DETAILED_INFORMATION actCtxInfo = nullptr;
+    SIZE_T bufferSize = 0;
+    (void)QueryActCtxW(0,
+        hActCtx,
+        nullptr,
+        ActivationContextDetailedInformation,
+        nullptr,
+        0,
+        &bufferSize);
+    RETURN_HR_IF(HRESULT_FROM_WIN32(GetLastError()), bufferSize == 0);
+
+    actCtxInfo = (PACTIVATION_CONTEXT_DETAILED_INFORMATION) new BYTE[bufferSize];
```

Comment:
Maybe use std::vector or `auto buffer = make_unique<BYTE[]>(bufferSize);` and then use `reinterpret_cast<PACTCTXDETAILEDINFO>(buffer.get())`

Reply:
(and if you do so, wrap the function with a function-level try to catch any failures.)